### PR TITLE
Clean up mdstore_oai, closes #27

### LINKED ISSUE #27: Clean up md_storeOAIInterface
There is a lot we can do to clean up `code_mdstore.mdstore_oai.mdstore_OAIInterface`. The docstring states that it implements the `ResumptionOAIPHM` interface, but it more accurately implements the `IOAI` interface [here](https://github.com/infrae/pyoai/blob/master/src/oaipmh/interfaces.py)

For starters
- Remove `_get_old_solr_getRecord`
- Remove `setAssociations`
- At the very least, rename `listStuff` to something more descriptive. It is more likely that we can remove `listStuff` and just implement the functionality into the two methods that call it.

### PR DESCRIPTION
**Pull request by runderwood**
Ticks remaining boxes in #27

### CODE REVIEW

File: coda/coda_mdstore/mdstore_oai.py [hunk 1]
```diff
@@ -95,14 +94,14 @@ def listIdentifiers(self, metadataPrefix=None, from_=None, until=None,
...
-        return self.listStuff(
+        return self.makeList(
             metadataPrefix=metadataPrefix, from_=from_,
             until=until, set=set, cursor=cursor, batch_size=batch_size,
             headersOnly=False
         )
 
-    def listStuff(self, metadataPrefix=None, from_=None, until=None,
-                  set=None, cursor=0, batch_size=10, headersOnly=True):
+    def makeList(self, metadataPrefix=None, from_=None, until=None,
```
Comment (somexpert) [inline coda/coda_mdstore/mdstore_oai.py]: I think I'd personally like to see a bit more descriptive name, or instead, a more descriptive docstring.

Comment (ldko) [inline coda/coda_mdstore/mdstore_oai.py]: I agree; if we are changing the name from `listStuff`, more description would be worthwhile.

Comment (runderwood) [inline coda/coda_mdstore/mdstore_oai.py]: I'm not sure how to name the method more descriptively. It's within the context of an OAI interface, so I feel like the sort of generic list it refers to within that context would be fairly discernible to a hypothetical reader. I can add a note about making generic lists.

Comment (somexpert) [inline coda/coda_mdstore/mdstore_oai.py]: Well, it gives more info than was previously there, so works for me. :thumbsup:

**Comment by ldko** [comment]
Do we need to keep `setAssociationList` here?
```
def makeDataRecord(
    bagObject,
    domain="",
    metadataPrefix="oai_dc",
    setAssociationList=[]
):
```

**Comment by runderwood** [comment]
@ldko No! Good catch.

**Pull request merged**